File: Code Generation/responses/python/claude/medium/apply-substitutions.py
from typing import List
import re

class Solution:
    def applySubstitutions(self, replacements: List[List[str]], text: str) -> str:
        mapping = {k: v for k, v in replacements}
        
        def resolve(key):
            val = mapping[key]
            return re.sub(r'%([A-Z])%', lambda m: resolve(m.group(1)), val)
        
        return re.sub(r'%([A-Z])%', lambda m: resolve(m.group(1)), text)

File: Code Generation/responses/cpp/claude/medium/apply-substitutions.cpp
#include <string>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    string applySubstitutions(vector<vector<string>> replacements, string text) {
        unordered_map<string, string> mp;
        for (auto& r : replacements) mp[r[0]] = r[1];
        
        function<string(const string&)> resolve = [&](const string& s) -> string {
            string result;
            int i = 0;
            while (i < s.size()) {
                if (s[i] == '%') {
                    int j = s.find('%', i + 1);
                    string key = s.substr(i + 1, j - i - 1);
                    result += resolve(mp[key]);
                    i = j + 1;
                } else {
                    result += s[i++];
                }
            }
            return result;
        };
        
        return resolve(text);
    }
};